### PR TITLE
Convert CI to GitHub Actions

### LINKED ISSUE #41: Convert CI to GitHub Actions
## Expected Behaviour

Convert CI to GitHub Actions

## Current Behaviour

We use Travis and need to move away

## Possible Solution

See the approach in https://github.com/openfaas/faas-netes/tree/master/.github

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

## Motivation and Context

This will fix #41.
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

There are two workflows defined for this project, one is "build" the other one is "publish".
* To be able to trigger the build workflow we need to push:
```sh
$ git push
```

* To be able to trigger the publish workflow we need to create a tag and push: 
```sh
$ git tag v9999
$ git push --tags
```
Screen Shot 2020-11-22 at 23 40 04
Screen Shot 2020-11-22 at 23 39 57

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@master
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }} 
+      - name: Docker meta
+        id: meta
+        uses: crazy-max/ghaction-docker-meta@v1
```
Comment (Waterdrips) [inline .github/workflows/publish.yaml:23]: I might be wrong, but i thought we are not allowing external actions.

Comment (developer-guy) [reply .github/workflows/publish.yaml:23]: Actually, I took this piece of code from the sample which is made by @LucasRoesler in the certifier project, https://github.com/LucasRoesler/certifier/actions/runs/377326951/workflow

Comment (Waterdrips) [reply .github/workflows/publish.yaml:23]: thats not been merged as far as i can see, the approach in https://github.com/openfaas/faas-cli/blob/master/.github/workflows/publish.yaml#L39-L54 seems like what we have settled on

Comment (developer-guy) [reply .github/workflows/publish.yaml:23]: Should I update my like the example above?

**Pull request closed without merging**